src/main/scala/nl/knaw/dans/easy/fedora2vault/EasyFedora2vaultApp.scala

```diff
@@ -15,10 +15,71 @@
  */
 package nl.knaw.dans.easy.fedora2vault
 
+import java.io.InputStream
+
+import better.files.File
+import com.yourmediashelf.fedora.client.FedoraClient
+import javax.naming.ldap.InitialLdapContext
+import nl.knaw.dans.bag.v0.DansV0Bag
+import nl.knaw.dans.easy.fedora2vault.Command.FeedBackMessage
+import nl.knaw.dans.easy.fedora2vault.FoXml.{ getEmd, _ }
+
 import scala.util.{ Success, Try }
+import scala.xml.{ Elem, XML }
+
+class EasyFedora2vaultApp(configuration: Configuration) {
+  lazy val fedoraProvider: FedoraProvider = new FedoraProvider(new FedoraClient(configuration.fedoraCredentials))
+  lazy val ldapContext: InitialLdapContext = new InitialLdapContext(configuration.ldapEnv, null)
+  private lazy val ldap = new Ldap(ldapContext)
 
-class EasyFedora2vaultApp(configuration: Configuration)  {
+  def simpleTransform(datasetId: DatasetId, outputDir: File): Try[FeedBackMessage] = {
 
+    def managedMetadataStream(foXml: Elem, streamId: String, bag: DansV0Bag, metadataFile: String) = {
+      managedStreamLabel(foXml, streamId)
+        .map { label =>
+          val extension = label.split("[.]").last
+          val bagFile = s"$metadataFile.$extension"
+          fedoraProvider.disseminateDatastream(datasetId, streamId)
+            .map(bag.addMetadataStream(bagFile))
+            .tried.flatten
+        }
+    }
 
+    def compareManifest(bag: DansV0Bag)(streamId: String): Try[Any] = {
+      fedoraProvider.disseminateDatastream(datasetId, streamId)
+        .map { inputStream: InputStream =>
+          val manifestst = bag.payloadManifests
+          Success(()) // TODO EASY-2678
```

Comment:
given the TODO, is this something that needs to be implemented properly later on? Because of the unused `inputStream` and `manifestst`?

Reply:
Indeed, though validating sha's was not explicitly specified it is my intention for another subtask of the main issue.